realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -282,6 +284,38 @@ public void refresh() {
     }
 
     /**
+     * Blocks the current thread until new changes to the Realm are available or {@link #stopWaitForChange}
+     * is called from another thread.
+     *
+     * @return {@code true} If the Realm was updated to the latest version, {@code false} if it was
+     * cancelled by calling {@link #stopWaitForChange}.
+     * @throws IllegalStateException if attempting to wait within a transaction or a Looper thread.
+     */
+    public boolean waitForChange() {
+        checkIfValid();
+        if (isInTransaction()) {
+            throw new IllegalStateException("Cannot wait for changes inside of a transaction.");
+        }
+        if (Looper.myLooper() != null) {
+            throw new IllegalStateException("Cannot wait for changes inside a Looper thread. Use RealmChangeListeners instead.");
+        }
+        boolean hasChanged = sharedGroupManager.getSharedGroup().waitForChange();
+        if (hasChanged) {
+            sharedGroupManager.advanceRead();
+        }
+        return hasChanged;
+    }
+
+    /**
+     * Makes {@link #waitForChange()} return {@code false} immediately. This method is threadsafe
+     * and should be called from another thread than the one that called {@link #waitForChange}.
+     * Calling {@code #stopWaitForChange()} on a Realm that is not waiting for changes has no effect.
+     */
+    public void stopWaitForChange() {
+        sharedGroupManager.getSharedGroup().stopWaitForChange();
```

Comment:
Since if you call `stopWaitForChange()` you want it to stop, which is exactly what happens (even though it goes throw `waitForChange()` to do it. We could solve it by adding a lock for it, but I don't think that would be needed?